primitives/ethereum-poa/src/lib.rs

```diff
@@ -344,6 +414,27 @@ pub fn public_to_address(public: &[u8; 64]) -> Address {
 	result
 }
 
+/// Verify ethereum merkle proof.
+fn verify_merkle_proof<T: AsRef<[u8]>>(expected_root: H256, items: impl Iterator<Item = T>) -> bool {
```

Comment:
For now proof = whole set of transactions. That's also true for tx receipts. We may try to migrate to minimal proof in future.